phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -958,13 +960,19 @@ private void addCoprocessors(byte[] tableName, TableDescriptorBuilder builder, P
                             opts.put(NonTxIndexBuilder.CODEC_CLASS_NAME_KEY, PhoenixIndexCodec.class.getName());
                             IndexRegionObserver.enableIndexing(builder, PhoenixIndexBuilder.class, opts, priority);
                         }
+                        if (newDesc.hasCoprocessor(Indexer.class.getName())) {
```

Comment:
Should we remove this before adding the new co-proc in the code-block right above? Since we are passing arounf the builder object to IndexRegionObserver it might be better to do cleanup first.